tools/entrypoint.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env sh
+# Script to run qdrant in docker container and handle contingencies, like OOM.
+# The functioning logic is as follows:
+# - If recovery mode is allowed, we check if qdrant was killed during initialization or not.
+#   - If it was killed during initialization, we remove run qdrant in recovery mode
+#   - If it was killed after initialization, do nothing and restart container
+# - If recovery mode is not allowed, we just restart container
+
+./qdrant $@
+
+EXIT_CODE=$?
+
+QDRANT_ALLOW_RECOVERY_MODE=${QDRANT_ALLOW_RECOVERY_MODE:-false}
+
+# Check that recovery mode is allowed
+if [ "$QDRANT_ALLOW_RECOVERY_MODE" = true ]; then
```

Comment:
Maybe we can do an early return here instead. It would make the script a bit less deep.